MeetupPlatformApi/Features/Authentication/ChangeUserCredentials/ChangeUserCredentialsFeature.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace MeetupPlatformApi.Features.Authentication.ChangeUserCredentials;
+
+using BCrypt.Net;
+using MeetupPlatformApi.Persistence.Context;
+using MeetupPlatformApi.Seedwork.WebApi;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+[ApiSection(ApiSections.Authentication)]
+public class ChangeUserCredentialsFeature : FeatureBase
+{
+    private readonly ApplicationContext context;
+
+    public ChangeUserCredentialsFeature(ApplicationContext context)
+    {
+        this.context = context;
+    }
+
+    [HttpPut("/api/users/me/credentials")]
+    [Authorize]
+    public async Task<IActionResult> ChangeCredentials([FromBody] UserCredentialsChangeDto credentialsChangeDto)
+    {
+        var user = await context.Users.SingleOrDefaultAsync(user => user.Id == CurrentUser.UserId);
+        var unallowedUsername = credentialsChangeDto.Username == user.Username ? false
```

Comment:
Consider renaming to `allowedUsername` (we also would need to invert the expression).